Library/Homebrew/cmd/test-bot.rb

```diff
@@ -651,7 +651,8 @@ def cleanup_before
         safe_system "git", "checkout", "-f", "master"
         # This will uninstall all formulae, as long as
         # HOMEBREW_REPOSITORY == HOMEBREW_PREFIX, which is true on the test bots
-        safe_system "git", "clean", "-ffdx", "--exclude=/Library/Taps/" unless ENV["HOMEBREW_RUBY"] == "1.8.7"
+        Tap.names.reject { |t| t == "homebrew/core" }.each { |s| safe_system "brew untap #{s}" }
```

Comment:
> If I'm reading the code of test-bot correctly, before_cleanup will be executed after installing the tap required for the run. Thus, running test-bot, e.g., for homebrew/versions will install this tap and then immediately untap it again. Right?

I'm half-asleep but AFAIK:

```
    def run
      cleanup_before
      begin
        download
        setup
        homebrew
        formulae.each do |f|
          formula(f)
        end
      ensure
        cleanup_after
      end
      check_results
    end
```

The `brew tap` happens during the `formula` stage? But potentially an issue for tap builds, perhaps. If I'm wrong I'll fix it in the morning.